core/src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala

```diff
@@ -765,6 +769,124 @@ class SubmissionSpec(_system: ActorSystem)
+    val sset = Entity(
+      "testset6",
+      "Sample_set",
+      Map(
+        AttributeName.withDefaultNS("samples") -> AttributeEntityReferenceList(
+          Seq(
+            AttributeEntityReference("Sample", "sample1"),
+            AttributeEntityReference("Sample", "sample2"),
+            AttributeEntityReference("Sample", "sample3"),
+            AttributeEntityReference("Sample", "sample4"),
+            AttributeEntityReference("Sample", "sample5"),
+            AttributeEntityReference("Sample", "sample6")
+          )
+        )
+      )
+    )
+
+    runAndWait(entityQuery.save(testData.workspace, sset))
+
+    val submissionRq = SubmissionRequest(
+      methodConfigurationNamespace = "dsde",
+      methodConfigurationName = "GoodMethodConfig",
+      entityType = Option(sset.entityType),
+      entityName = Option(sset.name),
+      expression = Option("this.samples"),
+      useCallCache = false,
+      deleteIntermediateOutputFiles = false,
+      deleteEntityType = Option(sset.entityType),
+      deleteEntityName = Option(sset.name)
+    )
+    val newSubmissionReport =
+      Await.result(submissionsService.createSubmission(testData.wsName, submissionRq), Duration.Inf)
+
+    assert(newSubmissionReport.workflows.size == 6)
+
+    checkSubmissionStatus(submissionsService, newSubmissionReport.submissionId)
+
+    val submission = runAndWait(submissionQuery.loadSubmission(UUID.fromString(newSubmissionReport.submissionId))).get
+    assert(submission.workflows.forall(_.status == WorkflowStatuses.Queued))
+
+    assertResult(None) {
+      runAndWait(entityQuery.get(testData.workspace, sset.entityType, sset.name))
+    }
+  }
+
+  it should "not delete entity if terraCreatedSetToDeleteAttribute not set even if preserveSet is set to false" in withSubmissionsService {
+    submissionsService =>
+      val submissionRq = SubmissionRequest(
+        methodConfigurationNamespace = "dsde",
+        methodConfigurationName = "GoodMethodConfig",
+        entityType = Option("Sample"),
+        entityName = Option("sample1"),
+        expression = None,
+        useCallCache = false,
+        deleteIntermediateOutputFiles = false,
+        deleteEntityType = Option("Sample"),
+        deleteEntityName = Option("sample1")
+      )
+      val newSubmissionReport =
+        Await.result(submissionsService.createSubmission(testData.wsName, submissionRq), Duration.Inf)
+
+      assert(newSubmissionReport.workflows.size == 1)
+
+      checkSubmissionStatus(submissionsService, newSubmissionReport.submissionId)
+
+      val submission = runAndWait(submissionQuery.loadSubmission(UUID.fromString(newSubmissionReport.submissionId))).get
+      assert(submission.workflows.forall(_.status == WorkflowStatuses.Queued))
+
+      assertResult(Some(testData.sample1)) {
+        runAndWait(entityQuery.get(testData.workspace, "Sample", "sample1"))
+      }
+  }
```

Comment:
help? I'm trying to understand why this test passes. What is the functional difference between this test and the test currently named "delete entity set if preserveSet is set to false"? I can see that this one passes but I think it should fail.

Reply:
ahhh … I think I understand. This submission is running against `testData.sample1`. However, other entities in `testData`, including `testData.sset1`, have a reference to `testData.sample1`. So, the code makes an attempt to delete `sample1` but that deletion fails and everything else continues on.